library/src/main/java/com/voysis/recorder/FileAudioSource.kt

```
@@ -26,11 +28,16 @@ class FileAudioSource(var wavFile: File? = null) : Source {
     }
 
     override fun read(buffer: ByteArray, i: Int, size: Int): Int {
-        bytesRead = inputStream?.read(buffer, 0, buffer.size)!!
+        try {
+            bytesRead = inputStream?.read(buffer, 0, buffer.size)!!
+        } catch (e: IOException) {
```

I don't agree with hiding this exception as an EOF read. The exception should bubble out of here so that the caller can properly respond to the exceptional condition. I realise doing it here may have made the job of handling an exception occurring easier in the existing implementation, but this is a definition problem for the future.

---

Good point. I handled it this way to account for lifecycle management. The user can back out of the app at any stage in which case destroy will need to be called on input stream which is being read from. I could use a state variable to check that the the input stream can be read from before calling read and also throw the IOException? This way an exception isn't thrown whenever the user backgrounds the app? open to suggestions.